debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/stackwalker/StackWalker.java

```diff
@@ -139,116 +154,165 @@ private static IStackWalker getImpl()
+			} catch (CorruptDataException ex) {
+				raiseCorruptDataEvent("CDE thrown extracting initial stack walk state. walkThread = " + walkThread.getHexAddress(), ex, true);
+				return StackWalkResult.STACK_CORRUPT;
+			}
+			return walkStackFrames(walkState);
+		}
+
+		private Object getField(Lookup lookup, Object obj, Class<?> refc, String name, Class<?> rtype) throws Throwable {
+			MethodHandle mh = lookup.findVirtual(refc, name, MethodType.methodType(rtype));
+			return mh.invoke(obj);
+		}
+
+		@Override
+		public StackWalkResult walkStackFrames(WalkState walkState, long continuationAddress) {
 			try {
-				return walkStackFrames(walkState, 
-						walkState.walkThread.sp(), 
-						walkState.walkThread.arg0EA(), 
-						walkState.walkThread.pc(), 
-						walkState.walkThread.literals(),
-						walkState.walkThread.entryLocalStorage());
+				Class<?> contClass = Class.forName("com.ibm.j9ddr.vm29.pointer.generated.J9VMContinuationPointer");
+				Lookup lookup = MethodHandles.lookup();
+				MethodHandle mhCast = lookup.findStatic(contClass, "cast", MethodType.methodType(contClass, long.class));
+				Object cont = mhCast.invoke(continuationAddress);
+
+				walkState.threadAddress = (long)getField(lookup, cont, contClass, "getAddress", long.class);
+				walkState.arg0EA = (UDATAPointer)getField(lookup, cont, contClass, "arg0EA", UDATAPointer.class);
+				walkState.pc = (U8Pointer)getField(lookup, cont, contClass, "pc", U8Pointer.class);
+				walkState.pcAddress = (PointerPointer)getField(lookup, cont, contClass, "pcEA", PointerPointer.class);
+				walkState.walkSP = (UDATAPointer)getField(lookup, cont, contClass, "sp", UDATAPointer.class);
+				walkState.literals  = (J9MethodPointer)getField(lookup, cont, contClass, "literals", J9MethodPointer.class);
+				walkState.privateFlags  = new UDATA(0);
+				walkState.j2iFrame = (UDATAPointer)getField(lookup, cont, contClass, "j2iFrame", UDATAPointer.class);
+				walkState.decompilationStack = (J9JITDecompilationInfoPointer)getField(lookup, cont, contClass, "decompilationStack", J9JITDecompilationInfoPointer.class);
+
+				final J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+				walkState.javaVM = vm;
+				J9VMEntryLocalStoragePointer stubELS = (J9VMEntryLocalStoragePointer)getField(lookup, vm,
+					J9JavaVMPointer.class, "entryLocalStorage", J9VMEntryLocalStoragePointer.class);
+				J9VMEntryLocalStoragePointerWrapper wrapperELS = J9VMEntryLocalStoragePointerWrapper.cast(stubELS);
+				wrapperELS.oldELS = (J9VMEntryLocalStoragePointer)getField(lookup, cont, contClass, "oldEntryLocalStorage", J9VMEntryLocalStoragePointer.class);
+				Class<?> ELSClass = Class.forName("com.ibm.j9ddr.vm29.pointer.generated.J9JITGPRSpillAreaPointer");
+				Object jitGPRs = getField(lookup, cont, contClass, "jitGPRs", ELSClass);
+				wrapperELS.jitGPRs = UDATAPointer.cast((long)getField(lookup, jitGPRs, ELSClass, "getAddress", long.class));
```

Comment:
I would prefer if you can set these fields in the wrapper's constructor, this avoid caller code having to modify the wrapper field directly, so future changes to the wrapper behaviour will be easier.